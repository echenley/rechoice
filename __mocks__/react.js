const React = require.requireActual('react')

React.Fragment = 'Fragment'

module.exports = React
