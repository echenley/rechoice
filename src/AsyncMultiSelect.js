// @flow

import * as React from 'react'
import Async from './Async'
import MultiSelect from './MultiSelect'
import type {
  FilterItemsMulti,
  LoadItems,
  Item,
  LoadingState,
  ItemRenderer,
  MenuRenderer,
  TagRenderer,
} from './types'

type Props = {
  disabled?: boolean,
  filterItems?: FilterItemsMulti,
  itemKey: string,
  loadingIndicator?: React.Node,
  loadingState?: LoadingState,
  loadItems: LoadItems,
  minimumInput?: number,
  onBlur?: () => mixed,
  onChange: (value: Array<Item>) => mixed,
  onFocus?: () => mixed,
  onInputChange?: (inputValue: string) => mixed,
  placeholder?: string,
  renderItem: ItemRenderer,
  renderMenu?: MenuRenderer,
  renderTag: TagRenderer,
  value: Array<Item>,
}

const AsyncMultiSelect = ({
  loadItems,
  onChange,
  onFocus,
  onInputChange,
  ...props
}: Props) => {
  return (
    <Async
      loadItems={loadItems}
      minimumInput={props.minimumInput}
      onChange={onChange}
      onFocus={onFocus}
      onInputChange={onInputChange}
    >
      {asyncProps => <MultiSelect {...props} {...asyncProps} />}
    </Async>
  )
}

export default AsyncMultiSelect
