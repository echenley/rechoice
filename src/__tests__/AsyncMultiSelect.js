import * as React from 'react'
import { shallow } from 'enzyme'
import AsyncMultiSelect from '../AsyncMultiSelect'

const renderComponent = props => {
  return shallow(
    <AsyncMultiSelect
      itemKey="id"
      loadItems={() => Promise.resolve({ items: [] })}
      loadingIndicator={<div className="loading-indicator" />}
      onChange={() => {}}
      onFocus={() => {}}
      onInputChange={() => {}}
      placeholder="Placeholder"
      renderItem={() => <div />}
      renderTag={item => <div>{item.id}</div>}
      value={[]}
      {...props}
    />
  )
}

describe('AsyncMultiSelect', () => {
  it('renders correctly', () => {
    expect(renderComponent().getElement()).toMatchSnapshot()
  })

  it('renders a MultiSelect child', () => {
    expect(
      renderComponent()
        .dive()
        .getElement()
    ).toMatchSnapshot()
  })
})
