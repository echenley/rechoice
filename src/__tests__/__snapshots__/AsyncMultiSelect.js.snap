// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsyncMultiSelect renders a MultiSelect child 1`] = `
<MultiSelect
  disabled={false}
  filterItems={[Function]}
  itemKey="id"
  items={Array []}
  loadingIndicator={
    <div
      className="loading-indicator"
    />
  }
  loadingState="success"
  minimumInput={0}
  onChange={[Function]}
  onFocus={[Function]}
  onInputChange={[Function]}
  placeholder="Placeholder"
  renderItem={[Function]}
  renderTag={[Function]}
  value={Array []}
/>
`;

exports[`AsyncMultiSelect renders correctly 1`] = `
<Async
  loadItems={[Function]}
  minimumInput={0}
  onChange={[Function]}
  onFocus={[Function]}
  onInputChange={[Function]}
>
  [Function]
</Async>
`;
